kurl_proxy/cmd/main.go

```diff
@@ -258,6 +266,12 @@ func getHttpsServer(upstream *url.URL, tlsSecretName string, secrets corev1.Secr
 
 		go func() {
 			<-c.Request.Context().Done()
+
+			if len(secret.StringData) == 0 {
+				secret.StringData = make(map[string]string)
+			}
+			secret.StringData["hostname"] = hostString
```

Comment:
I didn't realize you could set arbitrary keys in secret of type TLS. I guess the acceptAnonymousUploads flag didn't really have to be moved to annotations?

Reply:
The next change that's coming is to watch if the secret exists or not.  If not, then have user go through the TLS certs upload workflow.  This is as opposed to using the flag; so the flag may go away.  This way we have 3 key pieces of information stored in the Data section of the Secret.  And we just manage secret creation.